### PR TITLE
Update ReferralLinks

### PR DESCRIPTION
**Pull request by elian-kurtenkov**
Update blue.API to match the ReferralLinks service contract

### CODE REVIEW

File: src/Lykke.blue.Api/Lykke.blue.Api.csproj [hunk 1]
```diff
@@ -40,7 +40,7 @@
     <PackageReference Include="AutoMapper" Version="6.1.1" />
     <PackageReference Include="FluentValidation.AspNetCore" Version="7.2.1" />
     <PackageReference Include="Lykke.blue.Service.InspireStream.Client" Version="1.0.10-beta" />
-    <PackageReference Include="Lykke.blue.Service.ReferralLinks.Client" Version="1.0.8-beta" />
```
Comment (NovichikhinAlexey) [inline src/Lykke.blue.Api/Lykke.blue.Api.csproj:43]: beta in test env not possible

File: src/Lykke.blue.Api/Controllers/RefLinksController.cs [hunk 2]
```diff
@@ -132,69 +134,77 @@ public class RefLinksController : BluApiBaseController
...
-            var serviceRequest = request.ConvertToServiceModel();
-            serviceRequest.ClientId = _requestContext.ClientId;
+            return NotFound("Reserved for version 2");
+
+            //var serviceRequest = request.ConvertToServiceModel();
+            //serviceRequest.ClientId = _requestContext.ClientId;
 
-            var result = await ExecuteRefLinksMethod((p) => _referralLinksService.FinalizeRefLinkTransferWithHttpMessagesAsync(p), serviceRequest, request.LogMessage);
-            return result;         
+            //var result = await ExecuteRefLinksMethod((p) => _referralLinksService.FinalizeRefLinkTransferWithHttpMessagesAsync(p), serviceRequest, request.LogMessage);
```
Comment (NovichikhinAlexey) [inline src/Lykke.blue.Api/Controllers/RefLinksController.cs]: remove commented text, use git for history

File: src/Lykke.blue.Api/Lykke.blue.Api.csproj (hunk 1 above)
Comment (elian-kurtenkov) [reply src/Lykke.blue.Api/Lykke.blue.Api.csproj:43]: Don't have a non-beta yet. First version is in review.

File: src/Lykke.blue.Api/Controllers/ReferralLinksController.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+using Swashbuckle.SwaggerGen.Annotations;
+using System;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Lykke.blue.Api.Controllers
+{
+    [Route("api/referralLinks")]
+    [Authorize]
+    public class RefLinksController : BluApiBaseController
```
Comment (NovichikhinAlexey) [inline src/Lykke.blue.Api/Controllers/ReferralLinksController.cs]: file name ReferralLinksController.cs
please rename file or class

File: src/Lykke.blue.Api/Controllers/ReferralLinksController.cs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        [SwaggerOperation("GetReferralLinksStatisticsBySenderId")]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        [ProducesResponseType(typeof(GetReferralLinksStatisticsBySenderIdResponse), (int)HttpStatusCode.OK)]
+        public async Task<IActionResult> GetReferralLinksStatisticsBySenderId()
+        {
+            var result = await ExecuteRefLinksMethod((p) => _referralLinksService.GetReferralLinksStatisticsBySenderIdWithHttpMessagesAsync(p), new RefLinkStatisticsRequest { SenderClientId = _requestContext.ClientId }, "Reflink statistics requested");
+            return result;
+        }
+        
+        private async Task<IActionResult> ExecuteRefLinksMethod<U, T>(Func<U, Task<HttpOperationResponse<T>>> method, U request, string logMessage)
```
Comment (NovichikhinAlexey) [inline src/Lykke.blue.Api/Controllers/ReferralLinksController.cs]: resharper warning for "U", plase rename to "TU"

File: src/Lykke.blue.Api/Lykke.blue.Api.csproj (hunk 1 above)
Comment (NovichikhinAlexey) [reply src/Lykke.blue.Api/Lykke.blue.Api.csproj:43]: Before review this component, you must 
1. push to test Lykke.blue.Service.ReferralLinks this service
2. build not beta version of Lykke.blue.Service.ReferralLinks.Client
3. update in this project client library to not beta version
4. and make pull request for review

after revicew on test you push branch with correct client library

**Comment by filip-prime** [comment]
Rename RefLinksController to ReferralLinksController. Reanme genreic param to TU - resharper suggestion. Done.

**Review by NovichikhinAlexey — APPROVED** [review]

**Pull request merged**